### PR TITLE
[FC2Video] Add support, fix #102 ; Add custom header support on most of the download-related functions

### LINKED ISSUE #102: [Support Request] FC2Video
- http://video.fc2.com/en/content/20130212WcPLbYY1/?otag=1
- http://video.fc2.com/content/So%E3%80%80long%21%E3%80%80%E7%AC%AC1%E5%A4%9C%E3%80%8020130212/20130212WcPLbYY1/&tk=TWpVd01EazJNekU9

### PR DESCRIPTION
**Pull request by cnbeining**
How could this project survive without customizing Headers?

Now Able to call with

```
download_urls([video_url], title, ext, size, output_dir, merge=merge, headers = fake_headers)
```

and everywhere where you can use Faker.

```
python3 you-get   --debug  http://video.fc2.com/en/content/20151021bTVKnbEw
Site:       FC2Video
Title:      ハーメルン市街地
Type:       Flash video (video/x-flv)
Size:       37.01 MiB (38806848 Bytes)

Downloading ハーメルン市街地.flv ...
  2.7% (  1.0/37.0 MB) ├█─────────────────────────────────────┤[1/1]  896 kB/s
```

[Review on Reviewable](https://reviewable.io/reviews/soimort/you-get/785)

### CODE REVIEW

File: src/you_get/extractors/fc2video.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    }
+    api_base = 'http://video.fc2.com/ginfo.php?upid={upid}&mimi={mimi}'.format(upid = upid, mimi = makeMimi(upid))
+    html = get_content(api_base, headers=fake_headers)
+    
+    video_url = match1(html, r'filepath=(.+)&sec')
+    video_url = video_url.replace('&mid', '?mid')
+    
+    title = match1(html, r'&title=(.+)&seek_image')
+    
+    type, ext, size = url_info(video_url, headers=fake_headers)
```
Comment (soimort) [inline src/you_get/extractors/fc2video.py]: So `url_info` requires `fake_headers`?

I prefer not to have bunch of fake headers in individual extractors, only the minimal field(s) that just works would be sufficient. Will see if this doesn't break anything.

Comment (cnbeining) [inline src/you_get/extractors/fc2video.py]: It turns out that I have to use this header to make it work: the built-in one won't work.

So I really hate this PR for this involves so many changes. Making things hard to test.

Should I open an issue to talk through about this major change of ```common.py```?

Comment (soimort) [inline src/you_get/extractors/fc2video.py]: For me now, it's no good to risk introducing breaking changes to `common.py`.
**1**. I always wanted to rewrite the whole networking routines from scratch, having already `get_response`, `get_html`, `get_decoded_html`, `get_content`, `get_head`, `url_locations`, `url_save`, `url_save_chunked`, ... are a hell of a lot of chaos.
**2**. If you want to propose changes to those, `common.py` will end up even more unmaintainable.
**3**. I forgot what I just wanted to say, but don't be stressed about this; we're not in a hurry to implement custom headers, for this specific site, Things will become better when I find time to do **1**.

Comment (cnbeining) [inline src/you_get/extractors/fc2video.py]: Nice move.

I will withdraw this PR until further changes.

Thanks for your work.

**Comment by cnbeining** [comment]
This PR is closed for:

https://github.com/soimort/you-get/pull/785/files#r47165527

Will reintroduce after changes taken places.

**Comment by soimort** [comment]
In the future, please consider separating different changes (reformatting dict data, add headers support in `common.py`, add support for FC2 videos) into multiple commits; doing so will help us review and blame changes more easily.

Thanks!

**Comment by cnbeining** [comment]
I will. Thanks.

**Pull request merged**